keps/2724-topology-aware-schedling/README.md

```diff
@@ -0,0 +1,578 @@
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha (MVP):](#alpha-mvp)
+    - [Beta](#beta)
+    - [Stable](#stable)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Account usage by watching DaemonSet pods](#account-usage-by-watching-daemonset-pods)
+  - [Use label for workload](#use-label-for-workload)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces a mechanism, called Topology Aware Scheduling (TAS), to
+facilitate Job scheduling in Kueue by leveraging the information about the
+hierarchical organization of a datacenter.
+
+First, we start by observing that data centers have organizational units (like
+racks and blocks). VMs running within the same organizational unit have better
+network bandwidth than VMs on different units. Second, the units form
+a hierarchical structure - there are multiple nodes within a rack, and there are
+multiple racks within a block. We say that nods placed in different racks are
+more distant than nodes placed within the same rack. Similarly, nodes placed in
+different blocks are more distant than two nodes within the same block.
+
+Based on these observations we propose a convention to expose the information
+about the node placement in a datacenter hierarchy using node labels.
+
+We also propose a set of APIs for Kueue administrators and users to utilize this
+information in order to optimize the network throughput between the pods.
+
+## Motivation
+
+It is common that AI / ML workloads require a significant amount of pod-to-pod
+communication to make progress. Thus, it is important, for runtime and overall
+cost, to make sure there is a high throughput network connection between the
+running pods.
+
+For example, some workloads may run twice slower if there is a pod placed on
+a node which belongs to a different block. However, currently the end user of
+Kueue has no way to "require" that all pods of its workload run within the same
+block.
+
+### Goals
+
+- allow a user to express that a workload can schedule only if all of its pods
+  can be placed on nodes which are close together (within the same level of
+  hierarchy)
+- allow a user to express that a workload prefers to schedule all of its pods
+  within the same level of hierarchy, but automatically relax the constraint if
+  not possible.
+
+### Non-Goals
+
+- support MultiKueue at the management cluster
+- support Cluster Autoscaler (ProvisioningRequest in particular)
+
+The above features might be pursued in the future in follow up KEPs.
+
+## Proposal
+
+* Introduce convention to represent the topological hierarchy as node labels
+* Provide API to configure the list of node labels representing levels in the
+  hierarchy per Resource Flavor
+* Introduce a set of Job annotations which allow to describe requirements for
+  TAS
+
+### User Stories
```

Comment:
Maybe add examples for how the jobs look like for each story. And add one story for jobset.

Reply:
Good idea, added examples both for Job and JobSet.